packages/widgets/src/lib/components/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte

```diff
@@ -1,16 +1,47 @@
 <script lang="ts">
-	import { TASKS_DATA, type PipelineType } from "@huggingface/tasks";
+	import { widgetNoInference } from "../../stores.js";
+	import { TASKS_DATA } from "@huggingface/tasks";
+	import type { WidgetExample, WidgetExampleAttribute } from "@huggingface/tasks";
+	import type { WidgetProps, ExampleRunOpts } from "../types.js";
 	import { getPipelineTask } from "../../../../utils/ViewUtils.js";
 	import IconInfo from "../../..//Icons/IconInfo.svelte";
 	import IconLightning from "../../..//Icons/IconLightning.svelte";
 	import PipelineTag from "../../../PipelineTag/PipelineTag.svelte";
+	import WidgetExamples from "../WidgetExamples/WidgetExamples.svelte";
 
+	type TWidgetExample = $$Generic<WidgetExample>;
+
+	export let model: WidgetProps["model"];
 	export let noTitle = false;
 	export let title: string | null = null;
-	export let pipeline: PipelineType | undefined;
+	export let isLoading = false;
 	export let isDisabled = false;
+	export let applyWidgetExample: (sample: TWidgetExample, opts?: ExampleRunOpts) => void = () => {};
+	export let validateExample: (sample: WidgetExample) => sample is TWidgetExample = (
+		sample
+	): sample is TWidgetExample => true;
+	export let callApiOnMount: WidgetProps["callApiOnMount"] = false;
+	export let exampleQueryParams: WidgetExampleAttribute[] = [];
+
+	const pipeline = model?.pipeline_tag;
 
 	$: task = pipeline ? getPipelineTask(pipeline) : undefined;
+
+	$: examplesAll = getExamples(isDisabled);
+
+	function getExamples(isDisabled: boolean): TWidgetExample[] {
+		const examples = ((model?.widgetData ?? []) as TWidgetExample[])
+			.filter((sample) => validateExample(sample) && (!isDisabled || sample.output !== undefined))
+			.sort((sample1, sample2) => (sample2.example_title ? 1 : 0) - (sample1.example_title ? 1 : 0));
```

Comment:
No need to cast here
```suggestion
		const examples = (model?.widgetData ?? [])
			.filter((sample): sample is TWidgetExample => validateExample(sample) && (!isDisabled || sample.output !== undefined))
```